relay-general/src/store/transactions/processor.rs

```diff
@@ -354,13 +354,13 @@ impl Processor for TransactionsProcessor<'_> {
             &TransactionSource::Url | &TransactionSource::Sanitized
         ) && self.name_config.scrub_identifiers
         {
-            let changed = scrub_identifiers(&mut event.transaction)?;
-            if changed && self.name_config.mark_scrubbed_as_sanitized {
-                let source = &mut event
+            scrub_identifiers(&mut event.transaction)?;
+            if self.name_config.mark_scrubbed_as_sanitized {
+                event
                     .transaction_info
                     .get_or_insert_with(Default::default)
-                    .source;
-                source.set_value(Some(TransactionSource::Sanitized));
+                    .source
+                    .set_value(Some(TransactionSource::Sanitized));
```

Comment:
The code structure is starting to look misleading (not your fault). I think it should look something like this:

```rust
// This comes from a feature flag that already toggles both rules and patterns,
// so we might as well put it around both function calls.
if self.name_config.scrub_identifiers {
    self.apply_transaction_rename_rule(
        &mut event.transaction,
        event.transaction_info.value_mut(),
    )?;
    if matches!(
            event.get_transaction_source(),
            &TransactionSource::Url | &TransactionSource::Sanitized) {
            
        scrub_identifiers(&mut event.transaction)?;
            
        if self.name_config.mark_scrubbed_as_sanitized {
            // set as sanitized
        }
    }
}
```

Reply:
That snippet changes the behavior. You're including `apply_transaction_rename_rule` inside the `scrub_identifiers ` feature flag now, and I don't think we want to do that.

In my opinion, considering the two feature flags the code is okayish as it is. Once we remove the feature flags, we can consider moving things around.

Reply:
Ohh, I see. Good suggestion, thanks! Addressed in 6961c5e. I also had to add the flag in tests, so that the scrubbing would run: 5d847ab.